### PR TITLE
hive uri qr codes support

### LINKED ISSUE #2688: add support for hive-uri QR codes
QR code to recognize some hive-uri. https://gitlab.syncad.com/hive/hive-uri

### PR DESCRIPTION
**Pull request by aliseyalvi**
### What does this PR?
Added support for parsing hive uri based QR codes and decoding the codes into hive transactions.
Currently it only supports two transaction types i-e vote and transfer
It can extended to support other transaction types in future

### Issue number
fixes #2688 

### Screenshots/Video

https://github.com/ecency/ecency-mobile/assets/48380998/8500521e-25e2-4633-a160-522b7fa10d47

### CODE REVIEW

**Review by noumantahir — CHANGES_REQUESTED** [review]
Please add/test the following items....

- [ ] - preview action details before proceeding for signing, I remember there was a action modal before, perhaps you can put it back but with better formatting... following something similar to what [hivesigner ](https://hivesigner.com/sign/op/WyJ2b3RlIix7InZvdGVyIjoiX19zaWduZXIiLCJhdXRob3IiOiJkZW1vLmNvbSIsInBlcm1saW5rIjoiZGVtb2RlbW8iLCJ3ZWlnaHQiOiIxMDAwIn1d) does by formatting the operations array.
- [ ] 
- [ ] - Test uri signing with accounts logged in using hivesigner
- [ ] - Test hive uri scan with user signed out, because qr scanning is accessible even when logged out
- [ ] - Test signing while in-app pin code is active, pin code is required for active key based transactions, posting

**Comment by aliseyalvi** [comment]
@noumantahir PR updated. 

https://github.com/ecency/ecency-mobile/assets/48380998/73c3eed6-b311-4fbe-9d6e-a2354d8cdd1d

File: src/components/qrModal/qrModalView.tsx [hunk 1]
```diff
@@ -97,42 +114,145 @@ export const QRModal = ({}: QRModalProps) => {
...
+      signers: currentAccount.name,
+      preferred_signer: currentAccount.name,
+    });
+    const operations = get(tx, 'operations', []);
+    dispatch(
+      showActionModal({
+        title: intl.formatMessage({
+          id: 'qr.confirmTransaction',
+        }),
+        bodyContent: _checkOpsArray(operations) ? _renderActionModalBody(operations[0]) : null,
```
Comment (noumantahir) [inline src/components/qrModal/qrModalView.tsx:170]: @aliseyalvi according to this, if checkOpsArray fails, operation will still continue upon confirmation, only this time user will be confirming blindly, wouldn't it make is counter intuitive.. why exactly cannot we show multiple operations array....? or perhaps if multiple ops detected, we can let user know app do no currently support singing multiple operations in a single transaction yet.

**Comment by noumantahir** [comment]
preview of transaction confirmation for private key based users.

Screenshot 2023-08-17 at 7 04 33 PM

File: src/components/qrModal/qrModalView.tsx [hunk 2]
```diff
@@ -97,42 +114,145 @@ export const QRModal = ({}: QRModalProps) => {
...
+        showWebViewModal({
+          uri: uri,
+        }),
+      );
+      return;
+    }
+
+    const parsed = hiveuri.decode(uri);
+    // resolve the decoded tx and params to a signable tx
+    let { tx, signer } = hiveuri.resolveTransaction(parsed.tx, parsed.params, {
```
Comment (noumantahir) [inline src/components/qrModal/qrModalView.tsx:160]: Might be a duplication, resolve transaction also happening inside `handleHiveUriOperation` in dhive.js

**Review by noumantahir — CHANGES_REQUESTED** [review]
just two small iterations and we should good for this PR... 

https://github.com/ecency/ecency-mobile/pull/2713#discussion_r1297254357
https://github.com/ecency/ecency-mobile/pull/2713#discussion_r1297277954

File: src/components/qrModal/qrModalView.tsx (hunk 2 above)
Comment (aliseyalvi) [reply src/components/qrModal/qrModalView.tsx:160]: @noumantahir this is not duplication, both server the different purpose, above resolveTransaction method is being used to get the tranaction details and show in action modal, 2nd one is being used in dhive.js for signing transaction

File: src/components/qrModal/qrModalView.tsx (hunk 1 above)
Comment (aliseyalvi) [reply src/components/qrModal/qrModalView.tsx:181]: @noumantahir change request is done here https://github.com/ecency/ecency-mobile/pull/2713/commits/55069cd74a95726f9120972531399d6958f7f334

**Comment by aliseyalvi** [comment]
@noumantahir [this](https://github.com/ecency/ecency-mobile/pull/2713#issuecomment-1682351011) change is done 
https://github.com/ecency/ecency-mobile/pull/2713/commits/d7a7a347fa4c8f3c6f72cbaa9498c1056c909dae
https://github.com/ecency/ecency-mobile/pull/2713/commits/0d22a37d966b51d44ff3704fe9e0c4ac1a12971d

**Review by noumantahir — APPROVED** [review]
lgtm

File: src/config/locales/en-US.json [hunk 3]
```diff
@@ -1031,7 +1031,12 @@
     "open": "Open URL",
     "detected_url": "Detected URL",
     "unsupported_alert_title": "Unsupported URL!",
-    "unsupported_alert_desc": "Please scan a valid ecency url."
+    "unsupported_alert_desc": "Please scan a valid ecency url.",
+    "confirmTransaction": "Confirm transaction",
+    "approve": "Approve",
+    "cancel": "Cancel",
+    "multi_array_ops_alert": "Multiple operations array detected!",
+    "multi_array_ops_aler_desct": "Ecency currently do not multiple operations array"
```
Comment (feruzm) [inline src/config/locales/en-US.json:1039]: @aliseyalvi, @noumantahir This should probably say `Ecency does not support signing multiple operations, yet`

**Pull request merged**